internal/wrapnix/wrapper.sh.tmpl

```diff
@@ -14,19 +14,21 @@ fi
 {{- end }}
 
 {{/*
-We use ShellEnvHashKey to prevent doing shellenv if the correct environment is
-already set. (perf optimization)
 
 We use the guard to prevent infinite loop if something in shellenv causes 
 another wrapped binary to be called. The guard is specific to this project so shellenv
 could still cause another project's shellenv to be called.
 
 DO_NOT_TRACK=1 can be removed once we optimize segment to queue events.
+
+--omit-wrappers-from-path so that we do not invoke other bin-wrappers from
+this bin-wrapper. Instead, we directly invoke the binary from the nix store, which
+should be in PATH.
 */ -}}
 
-if [[ "${{ .ShellEnvHashKey }}" != "{{ .ShellEnvHash }}" ]] && [[ -z "${{ .ShellEnvHashKey }}_GUARD" ]]; then
```

Comment:
This makes me sad. It makes the happy (common) path slower. Thoughts on skipping wrapper creation if hash matches? That's a risky change, so I guess it also depends on how soon we think we'll have the prompt solution.

Reply:
its very marginally slower, not perceptible, and not a huge concern IMO. We'll be running this same `devbox shellenv` on each prompt-display in the alternate solution.

I'm not keen on skipping wrapper creation if hash matches, since it feels like a riskier change to make.